src/test/java/org/openrewrite/staticanalysis/RemoveExtraSemicolonsTest.java

```diff
@@ -165,4 +164,54 @@ void test() {
           )
         );
     }
+
+    @Issue("https://github.com/openrewrite/rewrite-static-analysis/issues/99")
+    @Test
+    void manySemicolonBeforeStatement() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              class Test {
+                  int test() {
+                      int a = 1; //first we set a to 1
+                      ;a = 2;//then we set a to 2
+                      a = 3;//then we set a to 3
+                      ;a = 4;;;;;//then we set a to 4
+                      ;a = 5;//then we set a to 5
+                      a = 6;;//then we set a to 6
+                      if (a == 6) { //if a is 6
+                        ;a = 7;;//then if a is 6 we set a to 7
+                      }
+                      ;;
+                      ;//next we set a to 8
+                      ;a = 8;
+                      return a;
+                      ;
+                      //we are done!
```

Comment:
Awesome!

Reply:
Maybe we could add in block comments `/* foo; */` just to be sure,
but otherwise I really like the variations you've applied here.